### DISCUSSION TITLE
Multiline statictext names

### DISCUSSION THREAD

**Discussion by softhorizons** [joonaspaakko/ScriptUI-Dialog-Builder-Joonas#99] [General]
The editor automatically splits multiline statictext into multiple statictext's inside a group, something like:
```
var staticTextOneTwo = grpMain.add("group"); 
    staticTextOneTwo.add("statictext", undefined, "One line", {name: "staticTextOneTwo"}); 
    staticTextOneTwo.add("statictext", undefined, "followed by another", {name: "staticTextOneTwo"}); 
```

However
- the group isn't named
- the individual statictext's have the same name

This makes it very inconvenient to access the auto-generated group and sub-elements. You really want to access the group as control.grpMain.staticTextOneTwo. I find myself manually creating a group of single-line statictext's to work around this.

It would be great if the editor emitted:
```
var staticTextOneTwo = grpMain.add("group", undefined, {name: "staticTextOneTwo"}); 
    staticTextOneTwo.add("statictext", undefined, "One line", {name: "statictext1"}); 
    staticTextOneTwo.add("statictext", undefined, "followed by another", {name: "statictext2"}); 
```
Though really, it might be cleaner to not emit names at all for the single-line statictexts since it makes more sense to access them as ...staticTextOneTwo.children[0, 1, ...]. How about it?

**Comment by joonaspaakko** [comment]
I fixed it #100

**Reply by softhorizons** [reply]
Thank you! Minor detail: why is grp.preferredSize.width emitted twice? Like this:

/*
Code for Import https://scriptui.joonas.me — (Triple click to select): 
{"activeId":1,"items":{"item-0":{"id":0,"type":"Dialog","parentId":false,"style":{"enabled":true,"varName":null,"windowType":"Dialog","creationProps":{"su1PanelCoordinates":false,"maximizeButton":false,"minimizeButton":false,"independent":false,"closeButton":true,"borderless":false,"resizeable":false},"text":"Dialog","preferredSize":[0,0],"margins":16,"orientation":"column","spacing":10,"alignChildren":["center","top"]}},"item-1":{"id":1,"type":"StaticText","parentId":0,"style":{"enabled":true,"varName":"grpMulti","creationProps":{},"softWrap":true,"text":"Here's a bit of text which will be wrapped","justify":"left","preferredSize":[100,0],"alignment":null,"helpTip":null}}},"order":[0,1],"settings":{"importJSON":true,"indentSize":false,"cepExport":false,"includeCSSJS":true,"showDialog":true,"functionWrapper":false,"afterEffectsDockable":false,"itemReferenceList":"None"}}
*/ 

// DIALOG
// ======
var dialog = new Window("dialog"); 
    dialog.text = "Dialog"; 
    dialog.orientation = "column"; 
    dialog.alignChildren = ["center","top"]; 
    dialog.spacing = 10; 
    dialog.margins = 16; 

var grpMulti = dialog.add("group", undefined , {name: "grpMulti"}); 
    grpMulti.getText = function() { var t=[]; for ( var n=0; n<grpMulti.children.length; n++ ) { var text = grpMulti.children[n].text || ''; if ( text === '' ) text = ' '; t.push( text ); } return t.join('\n'); }; 
    grpMulti.preferredSize.width = 100; 
    grpMulti.orientation = "column"; 
    grpMulti.alignChildren = ["left","center"]; 
    grpMulti.spacing = 0; 
    grpMulti.add("statictext", undefined, "Here's a bit of"); 
    grpMulti.add("statictext", undefined, "text which will"); 
    grpMulti.add("statictext", undefined, "be wrapped"); 
    grpMulti.preferredSize.width = 100; 

dialog.show();

**Reply by joonaspaakko** [reply]
It didn't really cause any issues, but since it was a simple thing, I fixed it. I also updated changes online.
